### ISSUE TITLE
Part 5 question

### ISSUE THREAD

**Issue by VictorioBerra** [ForEvolve/ForEvolve.Blog.Samples#5]
http://www.forevolve.com/en/articles/2017/08/25/design-patterns-web-api-service-and-repository-part-5/

How was the integration done before .NET Core?

**Comment by Carl-Hugo** [comment]
If you meant "how was the end-to-end integration tests over HTTP coded before .NET Core", the first thing that comes to my mind is: more painfully.

That said, if you have an ASP.NET 4.x Web API that you want to test, I'd recommend taking a look at [Postman](https://www.getpostman.com/) (or any similar tools) instead. This may save you some headaches. 

*Postman can also be paired with Newman; that you can use in your CI/CD pipeline to automate your Postman's tests during builds. It can also be used to generate documentation for your API.*

**Comment by VictorioBerra** [comment]
Thanks for your reply, I have not heard of Newman. This could be good for E2E tests. Sadly it wouldn't let me do more complex things like mocking my DbContext or drop my DB, migrate and seed before each run of all tests.

**Comment by Carl-Hugo** [comment]
> I have not heard of Newman

Newman is only the console version of Postman.

> Sadly it wouldn't let me do more complex things like mocking my DbContext or drop my DB, migrate and seed before each run of all tests.

True enough. 

Mostly the same thing can be achieved, here is a solution that I found that looks promising (I have not tried it tho): [Unit Testing / Integration Testing Web API with HttpClient in Visual Studio 2013](https://stackoverflow.com/questions/37475387/unit-testing-integration-testing-web-api-with-httpclient-in-visual-studio-2013#answer-37510032)

After a quick search, I also found these:

- [Unit Testing Controllers in ASP.NET Web API 2](https://docs.microsoft.com/en-us/aspnet/web-api/overview/testing-and-debugging/unit-testing-controllers-in-web-api)
- [Writing tests for an ASP.NET Web API service](https://blogs.msdn.microsoft.com/youssefm/2013/01/28/writing-tests-for-an-asp-net-web-api-service/) this one seems to use an in-memory http server as well.

I hope this helps
